src/db.rs

```diff
@@ -0,0 +1,37 @@
+use std::fmt::Error;
+
+pub struct Db {}
+
+impl Db {
+    pub fn new() -> Db {
+        Db {}
+    }
+
+    pub fn get_deployment_join_project(&self, uuid: String) -> DeploymentJoinProject {
```

Comment:
I think it might be better to return a tuple of `(struct Deployment, struct Project)`.

That means that you have to create those two new structs.

Also, rename this function to `get_deployment_with_project`.

Reply:
Done differently but done nonetheless.

I chose to rename the function to `get_deployment_and_project`, which return a struct consisting of `Deployment` and `Project` (hence the name).